### PR TITLE
Experiment with testing package distro

### PR DESCRIPTION
**Pull request by fingolfin**
Status of "failing" package tests: I've moved this overview [to an issue of its own](https://github.com/gap-system/PackageDistro/issues/155), so that I can merge this PR.

### CODE REVIEW

File: .github/workflows/test-all.yml [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            echo "No required binary depedencies to be installed"
+          fi
+
+      - name: "Download packages"
+        run: |
+          #curl -L -o $HOME/packages.tar.gz ${{ github.event.inputs.pkgarchiveurl || inputs.pkgarchiveurl }}
+          #ls -l $HOME/packages.tar.gz
+          #shasum -a 256 $HOME/packages.tar.gz
+          ## TODO: also download and test checksum???
+          _tools/download_packages.py */meta.json
```
Comment (fingolfin) [inline .github/workflows/test-all.yml:61]: Not sure how to best "download packages":
1. using `download_packages.py` has the advantage that we really test the packages in the commit this action runs on
2. downloading a `packages.tar.gz` from a given ensures we really test the tarball we later offer to users; but it requires a download URL for such a package

In practice, I envision that this is run from either ...
1. a PR workflow (to test whether a PR is ready to be merged),
2. upon a `push` to the `main` branch (to test a `packages.tar.gz` we just assembled),
3. as part of a scheduled cron job, to test for regressions introduced in GAP `master`

For the first scenario, it makes sense to use `download_packages.py`; for the other two scenarios, I'd say the second approach makes more sense. So perhaps both should be supported, by turning the `pkgarchiveurl` into an optional argument?

**Comment by fingolfin** [comment]
Weird: most package test jobs end with GAP segfaulting, despite the tests being successful? E.g.
```
Segmentation fault
   58400 ms (8815 ms GC) and 7.50GB allocated for smallgroups.tst
/home/runner/gap/gap(+0x75a84)[0x5604893e9a84]
/lib/x86_64-linux-gnu/libc.so.6(+0x46210)[0x7ff939960210]
-----------------------------------
total    206068 ms (26619 ms GC) and 27.6GB allocated
              0 failures in 6 files

#I  No errors detected while testing
```
Note how the segfault message is intermingled with GAP's output. Weird.

**Comment by cdwensley** [comment]
xmod-2.85 and kan-1.34 now released and (hopefully) no longer failing.  (Just commented out a few lines in the tests where output differed slightly between gap4r11 and gapdev.)

**Comment by fingolfin** [comment]
kan-1.34 indeed passes its tests now, thank you!

But xmod-2.85 still has diffs, see [this log](https://github.com/fingolfin/PackageDistro/runs/5528182898?check_suite_focus=true)

**Comment by cdwensley** [comment]
The problem test, tst/extra/d24.tst, works for me with the latest gapdev.  However the log you sent shows 8 versions of packages more recent than the ones installed on my system.  So it appears that one of these packages is causing the failure.  I have, for now, moved d24.tst to tst/failing; released xmod-2.86; and hope there are no further problems.

**Pull request merged**